File: src/module/quest/quest_protocol.erl
-module(quest_protocol).
-export([read/2, write/2]).
-include("quest.hrl").


read(11201, <<>>) ->
    {ok, []};

read(11202, <<QuestId:32>>) ->
    {ok, QuestId};

read(11203, <<QuestId:32>>) ->
    {ok, QuestId};

read(Code, Binary) ->
    {error, Code, Binary}.



write(11201, List) ->
    ListBinary = protocol:write_list(fun(#quest{quest_id = QuestId, number = Number, is_award = IsAward}) -> <<QuestId:32, Number:16, IsAward:8>> end, List),
    {ok, protocol:pack(11201, <<ListBinary/binary>>)};

write(11202, [Result, #quest{quest_id = QuestId, number = Number, is_award = IsAward}]) ->
    {ok, protocol:pack(11202, <<(protocol:text(11202, Result))/binary, QuestId:32, Number:16, IsAward:8>>)};

write(11203, Result) ->
    {ok, protocol:pack(11203, <<(protocol:text(11203, Result))/binary>>)};

write(Code, Content) ->
    {error, Code, Content}.

File: script/make/protocol/json/QuestProtocol.js
let questProtocol = {
    "read" : {
        "11201" : [],
        "11202" : [
            {"name" : "questId", "type" : "u8", "comment" : "任务ID", "explain" : []}
        ],
        "11203" : [
            {"name" : "questId", "type" : "u8", "comment" : "任务ID", "explain" : []}
        ]
    },
    "write" : {
        "11201" : [
            {"name" : "list", "type" : "list", "comment" : "任务列表", "explain" : [
                {"name" : "questId", "type" : "u32", "comment" : "任务ID", "explain" : []},
                {"name" : "number", "type" : "u16", "comment" : "当前数量", "explain" : []},
                {"name" : "award", "type" : "u8", "comment" : "是否领取奖励", "explain" : []}
            ]}
        ],
        "11202" : [
            {"name" : "result", "type" : "u8", "comment" : "结果", "explain" : []},
            {"name" : "questId", "type" : "u32", "comment" : "任务ID", "explain" : []},
            {"name" : "number", "type" : "u16", "comment" : "当前数量", "explain" : []},
            {"name" : "award", "type" : "u8", "comment" : "是否领取奖励", "explain" : []}
        ],
        "11203" : [
            {"name" : "result", "type" : "u8", "comment" : "结果", "explain" : []}
        ]
    }
};